src/lib.rs

```diff
@@ -350,23 +347,13 @@ pub(crate) unsafe fn rav1d_open(c_out: &mut *mut Rav1dContext, s: &Rav1dSettings
         ::core::mem::size_of::<Rav1dTaskContext>().wrapping_mul((*c).n_tc as usize),
     );
     if (*c).n_tc > 1 as c_uint {
-        if pthread_mutex_init(&mut (*c).task_thread.lock, 0 as *const pthread_mutexattr_t) != 0 {
-            return error(c, c_out, &mut thread_attr);
```

Comment:
In Rust, `Mutex`, `Condvar`, etc. can never fail on creation, right?

Reply:
Right